### DISCUSSION TITLE
How to represent orientation vectors for geometric primitives?

### DISCUSSION THREAD

**Discussion by CalebKornfein** [cityjson/specs#202] [Q&A] [answered]
Hello! I'm looking for some help to brainstorm the best way to be able to represent an orientation vector to represent "front" or "direction" for geometric primitives. I'd be curious why something like this is not represented in the CityJSON schema like it is in gLTF as well. Thank you for your help :).

**Comment by hugoledoux** [comment, marked as answer]
It is not there mostly because the geometric primitives are supposed to be stored according to the standard ISO 19107, which prescribes the orientation of surfaces for the different geometric primitives. See an overview of the rules for the geom primitives of CityJSON: https://val3dity.readthedocs.io/2.5.0/definitions.html

But if you really want to store the normal of a surface, you can store it as an attribute, see [here](https://www.cityjson.org/specs/2.0.1/#semantics-of-geometric-primitives) how the semantics is stored, but also other attributes can be stored. So you could add a property `"normal": [0.0, 0.0, 1.0],` for instance

**Reply by CalebKornfein** [reply]
Thank you for your helpful answer! To make sure I am understanding correctly, this would not be a CityJSON attribute (stored at the per-City Object level) but rather a semantic extension (which can be stored at the geometric primitive level)?

**Reply by hugoledoux** [reply]
Correct.
